### ISSUE TITLE
您好，我在使用数据集训练的时候，发现迭代次数与batchsize，和训练集的图片数量对不上，您知道为什么吗？

### ISSUE THREAD

**Issue by zehaojiang** [zcablii/SARDet_100K#57]
这是我的
coco_detection.py 的配置:
# dataset settings
dataset_type = 'CocoDataset'
data_root = 'data/coco/'

# Example to use different file client
# Method 1: simply set the data root and let the file I/O module
# automatically infer from prefix (not support LMDB and Memcache yet)

# data_root = 's3://openmmlab/datasets/detection/coco/'

# Method 2: Use `backend_args`, `file_client_args` in versions before 3.0.0rc6
# backend_args = dict(
#     backend='petrel',
#     path_mapping=dict({
#         './data/': 's3://openmmlab/datasets/detection/',
#         'data/': 's3://openmmlab/datasets/detection/'
#     }))
backend_args = None

train_pipeline = [
    dict(type='LoadImageFromFile', backend_args=backend_args),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(type='Resize', scale=(800, 800), keep_ratio=True),
    dict(type='RandomFlip', prob=0.5),
    dict(type='PackDetInputs')
]
test_pipeline = [
    dict(type='LoadImageFromFile', backend_args=backend_args),
    dict(type='Resize', scale=(800, 800), keep_ratio=True),
    # If you don't have a gt annotation, delete the pipeline
    dict(type='LoadAnnotations', with_bbox=True),
    dict(
        type='PackDetInputs',
        meta_keys=('img_id', 'img_path', 'ori_shape', 'img_shape',
                   'scale_factor'))
]
train_dataloader = dict(
    batch_size=2,
    num_workers=2,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=True),
    batch_sampler=dict(type='AspectRatioBatchSampler'),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        ann_file='Annotations/train.json',
        data_prefix=dict(img='train/'),
        filter_cfg=dict(filter_empty_gt=True, min_size=32),
        pipeline=train_pipeline,
        backend_args=backend_args))
val_dataloader = dict(
    batch_size=1,
    num_workers=2,
    persistent_workers=True,
    drop_last=False,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        ann_file='Annotations/val.json',
        data_prefix=dict(img='val/'),
        test_mode=True,
        pipeline=test_pipeline,
        backend_args=backend_args))
test_dataloader = val_dataloader

val_evaluator = dict(
    type='CocoMetric',
    ann_file=data_root + 'Annotations/val.json',
    metric='bbox',
    format_only=False,
    backend_args=backend_args)

test_dataloader = dict(
    batch_size=1,
    num_workers=2,
    persistent_workers=True,
    drop_last=False,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        ann_file='Annotations/test.json',
        data_prefix=dict(img='test/'),
        test_mode=True,
        pipeline=test_pipeline,
        backend_args=backend_args))

test_evaluator = dict(
    type='CocoMetric',
    ann_file=data_root + 'Annotations/test.json',
    metric='bbox',
    classwise = True,
    format_only=False,
    backend_args=backend_args)

这是schedule_1x.py的配置
# training schedule for 1x
train_cfg = dict(type='EpochBasedTrainLoop', max_epochs=12, val_interval=1)
val_cfg = dict(type='ValLoop')
test_cfg = dict(type='TestLoop')

# learning rate
param_scheduler = [
    dict(
        type='LinearLR', start_factor=0.001, by_epoch=True, begin=0, end=500),
    dict(
        type='MultiStepLR',
        begin=0,
        end=12,
        by_epoch=True,
        milestones=[8, 11],
        gamma=0.1)
]

# optimizer
optim_wrapper = dict(
    type='OptimWrapper',
    optimizer=dict(type='SGD', lr=0.02, momentum=0.9, weight_decay=0.0001))

# Default setting for scaling LR automatically
#   - `enable` means enable scaling LR automatically
#       or not by default.
#   - `base_batch_size` = (8 GPUs) x (2 samples per GPU).
auto_scale_lr = dict(enable=False, base_batch_size=16)

这是default_runtime.py的配置
default_scope = 'mmdet'

default_hooks = dict(
    timer=dict(type='IterTimerHook'),
    logger=dict(type='LoggerHook', interval=50),
    param_scheduler=dict(type='ParamSchedulerHook'),
    checkpoint=dict(type='CheckpointHook', interval=1),
    sampler_seed=dict(type='DistSamplerSeedHook'),
    visualization=dict(type='DetVisualizationHook'))

env_cfg = dict(
    cudnn_benchmark=False,
    mp_cfg=dict(mp_start_method='fork', opencv_num_threads=0),
    dist_cfg=dict(backend='nccl'),
)

vis_backends = [dict(type='LocalVisBackend')]
visualizer = dict(
    type='DetLocalVisualizer', vis_backends=vis_backends, name='visualizer')
log_processor = dict(type='LogProcessor', window_size=50, by_epoch=True)

log_level = 'INFO'
load_from = None
resume = False

这是部分的训练日志：
04/25 16:09:11 - mmengine - INFO - Epoch(train)  [1][ 5200/30644]  lr: 2.0000e-05  eta: 1 day, 4:43:48  time: 0.2826  data_time: 0.0025  memory: 2375  loss: 0.1771  loss_rpn_cls: 0.0309  loss_rpn_bbox: 0.0103  loss_cls: 0.0661  acc: 96.5820  loss_bbox: 0.0698
04/25 16:09:25 - mmengine - INFO - Epoch(train)  [1][ 5250/30644]  lr: 2.0000e-05  eta: 1 day, 4:43:21  time: 0.2817  data_time: 0.0028  memory: 2376  loss: 0.2035  loss_rpn_cls: 0.0409  loss_rpn_bbox: 0.0145  loss_cls: 0.0713  acc: 96.7773  loss_bbox: 0.0768
04/25 16:09:40 - mmengine - INFO - Epoch(train)  [1][ 5300/30644]  lr: 2.0000e-05  eta: 1 day, 4:42:57  time: 0.2824  data_time: 0.0026  memory: 2376  loss: 0.2077  loss_rpn_cls: 0.0425  loss_rpn_bbox: 0.0148  loss_cls: 0.0712  acc: 98.4375  loss_bbox: 0.0793
04/25 16:09:54 - mmengine - INFO - Epoch(train)  [1][ 5350/30644]  lr: 2.0000e-05  eta: 1 day, 4:42:34  time: 0.2826  data_time: 0.0029  memory: 2376  loss: 0.2233  loss_rpn_cls: 0.0476  loss_rpn_bbox: 0.0174  loss_cls: 0.0784  acc: 97.7539  loss_bbox: 0.0799
04/25 16:10:08 - mmengine - INFO - Epoch(train)  [1][ 5400/30644]  lr: 2.0000e-05  eta: 1 day, 4:42:10  time: 0.2824  data_time: 0.0027  memory: 2376  loss: 0.2257  loss_rpn_cls: 0.0399  loss_rpn_bbox: 0.0164  loss_cls: 0.0806  acc: 98.4375  loss_bbox: 0.0889
04/25 16:10:22 - mmengine - INFO - Epoch(train)  [1][ 5450/30644]  lr: 2.0000e-05  eta: 1 day, 4:41:50  time: 0.2832  data_time: 0.0027  memory: 2376  loss: 0.2095  loss_rpn_cls: 0.0391  loss_rpn_bbox: 0.0164  loss_cls: 0.0722  acc: 98.7305  loss_bbox: 0.0819

训练日志显示第1个trainEpoch，总的迭代次数是30644，我的batchsize设置是2，计算的图片是61288张，而训练集train中一共是94493张图片

**Comment by zehaojiang** [comment]
已经解决了，因为filter_cfg=dict（filter_empty_gt=True， min_size=32）这个把一部分数据给过滤掉了
